src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs

```diff
@@ -0,0 +1,220 @@
+﻿namespace ServiceControl.Audit.Persistence.SqlServer
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+    using Dapper;
+    using Infrastructure;
+    using NServiceBus.CustomChecks;
+    using NServiceBus.Logging;
+    using ServiceControl.Audit.Auditing;
+    using ServiceControl.Audit.Auditing.MessagesView;
+    using ServiceControl.Audit.Monitoring;
+    using ServiceControl.SagaAudit;
+
+    //TODO - not done as not sure on the table structure required here
+    class SqlDbAuditDataStore : IAuditDataStore
+    {
+        readonly SqlDbConnectionManager connectionManager;
+
+        public SqlDbAuditDataStore(SqlDbConnectionManager connectionManager)
+        {
+            this.connectionManager = connectionManager;
+        }
+
+        public async Task<QueryResult<IList<MessagesView>>> GetMessages(bool includeSystemMessages, PagingInfo pagingInfo, SortInfo sortInfo)
+        {
+            var messagesViews = new List<MessagesView>();
+
+            await connectionManager.Perform(async connection =>
+            {
+                var rows = await connection.QueryAsync("SELECT * FROM [MessagesView]").ConfigureAwait(false);
+
+                foreach (var row in rows)
+                {
+                    messagesViews.Add(new MessagesView
+                    {
+                        BodySize = row.BodySize,
+                        BodyUrl = row.BodyUrl,
+                        ConversationId = row.ConversationId,
+                        CriticalTime = row.CriticalTime,
+                        DeliveryTime = row.DeliveryTime,
+                        Id = row.Id,
+                        InstanceId = row.InstanceId,
+                        IsSystemMessage = row.IsSystemMessage,
+                        MessageId = row.MessageId,
+                        MessageIntent = row.MessageIntent,
+                        MessageType = row.MessageType,
+                        ProcessedAt = row.ProcessedAt,
+                        ProcessingTime = row.ProcessingTime,
+                        Status = row.Status,
+                        TimeSent = row.TimeSent
+                        //Headers = row.Headers,
+                        //InvokedSagas = row.InvokedSagas,
+                        //OriginatesFromSaga = row.OriginatesFromSaga,
+                        //ReceivingEndpoint = row.ReceivingEndpoint,
+                        //SendingEndpoint = row.SendingEndpoint
```

Comment:
Do we need these?

Reply:
It's what's left to do so I've added TODO comments above those to make it clear